<?xml version="1.0" encoding="UTF-8" ?>
<documentation title="PHP Constants" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../../../../tests/documentation.xsd">
    <standard>
    <![CDATA[
    The <em>true</em>, <em>false</em> and <em>null</em> constants must always be uppercase.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: uppercase constants">
        <![CDATA[
if ($var === <em>FALSE</em> || $var === <em>NULL</em>) {
    $var = <em>TRUE</em>;
}
        ]]>
        </code>
        <code title="Invalid: lowercase constants">
        <![CDATA[
if ($var === <em>false</em> || $var === <em>null</em>) {
    $var = <em>true</em>;
}
        ]]>
        </code>
    </code_comparison>
</documentation>
