<?xml version="1.0" encoding="UTF-8" ?>
<documentation title="Opening Brace in Function Declarations" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../../../../tests/documentation.xsd">
    <standard>
    <![CDATA[
    Function declarations follow the "BSD/Allman style". The function brace is on the line following the function declaration and is indented to the same column as the start of the function declaration.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: brace on next line">
        <![CDATA[
function fooFunction($arg1, $arg2 = '')
<em>{</em>
    ...
}
        ]]>
        </code>
        <code title="Invalid: brace on same line">
        <![CDATA[
function fooFunction($arg1, $arg2 = '') <em>{</em>
    ...
}
        ]]>
        </code>
    </code_comparison>
</documentation>
