<documentation title="For Loops With Function Calls in the Test">
    <standard>
    <![CDATA[
    For loops should not call functions inside the test for the loop when they can be computed beforehand.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: A for loop that determines its end condition before the loop starts.">
        <![CDATA[
for ($i = 0, <em>$end = count($array)</em>; $i < $end; $i++) {
    echo "{$i}: {$array[$i]}\n";
}
        ]]>
        </code>
        <code title="Invalid: A for loop that unnecessarily computes the same value on every iteration.">
        <![CDATA[
for ($i = 0; $i &lt; <em>count($array)</em>; $i++) {
    echo "{$i}: {$array[$i]}\n";
}
        ]]>
        </code>
    </code_comparison>
</documentation>
