<documentation title="Function Call Signature">
    <standard>
    <![CDATA[
    Checks that the function call format is correct.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Correct spacing used.">
        <![CDATA[
foo<em></em>(<em></em>$bar, $baz<em></em>) {
}
        ]]>
        </code>
        <code title="Invalid: Invalid spacing used.">
        <![CDATA[
foo<em> </em>(<em> </em>$bar, $baz<em> </em>) {
}
        ]]>
        </code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: Correct number of spaces used for indent.">
        <![CDATA[
foo(
<em>    </em>$bar,
<em>    </em>$baz
) {
}
        ]]>
        </code>
        <code title="Invalid: Incorrect number of spaces are used for indent.">
        <![CDATA[
foo(
<em>  </em>$bar,
<em>      </em>$baz
) {
}
        ]]>
        </code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: Correct closing parenthesis.">
        <![CDATA[
foo(
    $bar,
    $baz
<em>)</em> {
}
        ]]>
        </code>
        <code title="Invalid: Incorrect closing parenthesis.">
        <![CDATA[
foo(
    $bar,
    $baz<em>)</em> {
}
        ]]>
        </code>
    </code_comparison>
</documentation>
